collector/kubernetes.go

```diff
@@ -0,0 +1,133 @@
+package collector
+
+import (
+	"context"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// KubernetesCollector collects metrics about Kubernetes clusters
+type KubernetesCollector struct {
+	logger  log.Logger
+	errors  *prometheus.CounterVec
+	client  *godo.Client
+	timeout time.Duration
+
+	Up        *prometheus.Desc
+	Count     *prometheus.Desc
+	NodePools *prometheus.Desc
+	Nodes     *prometheus.Desc
+}
+
+// NewKubernetesCollector returns a new KubernetesCollector
+func NewKubernetesCollector(logger log.Logger, errors *prometheus.CounterVec, client *godo.Client, timeout time.Duration) *KubernetesCollector {
+	errors.WithLabelValues("kubernetes").Add(0)
+
+	// Version refers to the upstream Kubernetes version as well as the DigitalOcean revision
+	clusterLabels := []string{"id", "name", "region", "version"}
```

Comment:
I'm actually not sure about the version. It's most likely fine, but I feel like most people get this via kube-state-metrics anyway.
WDYT?